### PR TITLE
speed up computation of `log_m_k` and `log_s_k`

### PR DESCRIPTION
**Pull request by borea17**
The next scope and segmentation mask in log units can be computed more efficiently using the definition of LogSigmoid:

Typically, we would use a sigmoid layer to transform the logits output (i.e., `alpha_logits`) into probabilities:

image

Using LogSigmoid leads to 

image

It follows from the model description

image

This aligns with [Burgess et al. (2019)](https://arxiv.org/abs/1901.11390): 
> Both log alpha_k and log (1 - alpha_k) are computed directly in log units from the logits (using the log softmax operation).

(Appendix B.2)

### CODE REVIEW

**Review by baudm — COMMENTED** [review]
Hi, thanks for the PR. How much improvement are we talking about here? I've looked at the code and I don't think you even tried running it. `networks.py:796` would trigger a `NameError` since you referenced `log_scope` instead of `log_s_k`. Frankly, I doubt that there would be a significant speed increase in real-life usage (i.e. no artificial loops running just the concerned lines of code).

For easier understanding, I organized the code to closely mimic how the idea is presented in the paper. Thus, the `log_alpha_k` term as output of the Attention Network instead of directly outputting the next scope and mask. However, I like your insight on directly using the logits. This would eliminate the log -> exp -> log conversion, as well as the need for `self.eps`, and would lead to a more numerically stable operation. I might just implement it directly without drastically changing the Attention Network's API.
